include/zephyr/net/mdns_responder.h

```diff
@@ -0,0 +1,27 @@
+/** @file
+ * @brief mDNS responder API
+ *
+ * This file contains the mDNS responder API. These APIs are used by the
+ * to register mDNS records.
+ */
+
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <zephyr/net/dns_sd.h>
+
+/**
+ * @brief Register continuous memory of @ref dns_sd_rec records.
+ *
+ * mDNS responder will start with iteration over mDNS records registered using
+ * @ref DNS_SD_REGISTER_SERVICE (if any) and then go over external records.
+ *
+ * @param records An array of mDNS records
```

Comment:
Please add documentation here about the fact that we are storing and using the user supplied pointer directly (not copying it), so the pointer must point to a memory that is valid for the time the records are being used. Typically this would mean that the records pointer should not be allocated from stack.

Reply:
Done